examples/booleansearchparser.py

```diff
@@ -0,0 +1,486 @@
+#-*- coding: utf-8 -*-
+# vim:fileencoding=utf-8
+"""
+Boolean Search query parser (Based on searchparser: https://github.com/pyparsing/pyparsing/blob/master/examples/searchparser.py)
+
+version 2018-07-22
+
+This search query parser uses the excellent Pyparsing module 
+(http://pyparsing.sourceforge.net/) to parse search queries by users.
+It handles:
+
+* 'and', 'or' and implicit 'and' operators;
+* parentheses;
+* quoted strings;
+* wildcards at the end of a search term (help*);
+* wildcards at the begining of a search term (*lp);
+* non-western languages
+
+Requirements:
+* Python
+* Pyparsing
+
+
+CONTRIBUTORS:
+- Guiem Bosch
+- Francesc Garcia
+
+
+SAMPLE USAGE:
+from booleansearchparser import BooleanSearchParser
+bsp = BooleanSearchParser()
+text = u"wildcards at the begining of a search term "
+exprs= [
+    u"*cards and term", #True
+    u"wild* and term",  #True
+    u"not terms",       #True
+    u"terms or begin",  #False
+]
+for expr in exprs:
+    print bsp.match(text,expr)
+
+#non-western samples
+text = u"안녕하세요, 당신은 어떠세요?"
+exprs= [
+    u"*신은 and 어떠세요", #True
+    u"not 당신은",       #False
+    u"당신 or 당",  #False
+]
+for expr in exprs:
+    print bsp.match(text,expr)
```

Comment:
I would suggest using `from __future__ import print_function` and changing this over to be Python 2/3 compatible

Reply:
Sure, shall I add the changes o can you add it yourself?

Reply:
I'm sorry but I don't think I can edit your pull request.

Reply:
I've added your suggestion : https://github.com/xecgr/pyparsing/commit/f07b5776df171d3572e616644b48748dee320c18